### PR TITLE
Initial drafting of Storage Hierarchies section

### PR DESCRIPTION
**Pull request by awoods**
Resolves: https://github.com/OCFL/spec/issues/123

### CODE REVIEW

File: draft/spec/index.html [hunk 1]
```diff
@@ -746,6 +754,38 @@ <h2>Root Structure</h2>
...
+          <a>OCFL Object Root</a>s MUST be stored as direct children of a containing <a>OCFL Storage Root</a> or as the
+          terminal resource at the end of a directory storage hierarchy.
+        </p>
+        <p>
+          A common practice is to use a unique identifier scheme to compose this storage hierarchy, typically arranged
+          according to some form of the [[PairTree]] specification. Irrespective of the pattern chosen for the storage
+          hierarchies, the following restrictions apply:
+        </p>
+        <ol>
+          <li>Storage hierarchies MUST NOT include files within interim directories</li>
```
[the remark this answers is not in the document]
Comment (awoods) [reply draft/spec/index.html]: This is intended to mean that if you have a directory hierarchy, there can not be files (or OCFL Objects, for that matter) interspersed throughout that hierarchy.

File: draft/spec/index.html [hunk 2]
```diff
@@ -746,6 +754,38 @@ <h2>Root Structure</h2>
...
+          <li>Storage hierarchies MUST be terminated by OCFL Object Roots</li>
+          <li>Storage hierarchies within the same OCFL Storage Root SHOULD share the same layout pattern</li>
+          <li>Storage hierarchies within the same OCFL Storage Root SHOULD NOT include mixed dispositions (zip,
```
[the remark this answers is not in the document]
Comment (awoods) [reply draft/spec/index.html]: The difference in these bullets may be subtle, but testable.
> Storage hierarchies within the same OCFL Storage Root SHOULD share the same layout pattern

This is intended to mean that if a directory hierarchy is used, the pattern is consistent.

> Storage hierarchies within the same OCFL Storage Root SHOULD NOT include mixed dispositions

This is intended to mean that it is invalid to intermix directory hierarchies, top-level Objects and zips.

[the remark this answers is not in the document]
Comment (zimeon) [reply draft/spec/index.html]: OK, I see the point but did not understand that from the text. We should not use different terms "directory hierarchy" and "disposition" to mean pretty much the same thing. I propose: 

> * The storage hierarchy under an OCFL Storage Root SHOULD use just one layout pattern.
> * The storage hierarchy under an OCFL Storage Root SHOULD consistently use either a directory hierarchy of OCFL objects, top-level OCFL objects, or OCFL Object ZIPs.

(and I think we also need a section describing in more detail what we mean by an OCFL Object ZIP... currently mentioned only in definition of OCFL Object Root... #181)

File: draft/spec/index.html [hunk 3]
```diff
@@ -755,26 +755,26 @@ <h2>Root Structure</h2>
...
+          hierarchy or as direct children of a containing <a>OCFL Storage Root</a>.
         </p>
         <p>
           A common practice is to use a unique identifier scheme to compose this storage hierarchy, typically arranged
           according to some form of the [[PairTree]] specification. Irrespective of the pattern chosen for the storage
           hierarchies, the following restrictions apply:
         </p>
         <ol>
-          <li>Storage hierarchies MUST NOT include files within interim directories</li>
+          <li>Storage hierarchies MUST NOT include files within intermediate directories</li>
```
[the remark this answers is not in the document]
Comment (awoods) [reply draft/spec/index.html]: Lines 758,759 say where OCFL Objects must be located. This line (767) says that no other files are allowed in the storage hierarchy.

File: draft/spec/index.html [hunk 4]
```diff
@@ -755,26 +755,26 @@ <h2>Root Structure</h2>
...
           <li>Storage hierarchies MUST be terminated by OCFL Object Roots</li>
```
[the remark this answers is not in the document]
Comment (awoods) [reply draft/spec/index.html]: empty leaf directory -> invalid
empty storage root -> potentially valid (Is it a "Storage hierarchy" with no OCFL Objects?)

**Pull request merged**